lib/chef/win32/file.rb

```diff
@@ -155,6 +155,27 @@ def self.verify_links_supported!
         end
       end
 
+      def self.file_access_check(path, desired_access)
+        security_descriptor = Chef::ReservedNames::Win32::Security.get_file_security(path)
+        token_rights = Chef::ReservedNames::Win32::Security::TOKEN_IMPERSONATE | 
+                       Chef::ReservedNames::Win32::Security::TOKEN_QUERY |
+                       Chef::ReservedNames::Win32::Security::TOKEN_DUPLICATE | 
+                       Chef::ReservedNames::Win32::Security::STANDARD_RIGHTS_READ
+        token = Chef::ReservedNames::Win32::Security.open_process_token(
+          Chef::ReservedNames::Win32::Process.get_current_process,
+          token_rights)
+        duplicate_token = token.duplicate_token(:SecurityImpersonation)
```

Comment:
Take a look at https://github.com/chef/chef/blob/master/lib/chef/win32/handle.rb#L35
It seems like it takes care of freeing

The reason for duplicate token is because AccessCheck cannot use the primary token. It demands an impersonation token:

```
Chef::Exceptions::Win32APIError: An attempt has been made to operate on an impersonation token by a thread that is not currently impersonating a client.
---- Begin Win32 API output ----
System Error Code: 1309
System Error Message: An attempt has been made to operate on an impersonation token by a thread that is not currently impersonating a client.
---- End Win32 API output ----
```